frontend/lib/resolution/scope-queries.cpp

```diff
@@ -1278,12 +1499,11 @@ static bool lookupInScopeViz(Context* context,
       config |= LOOKUP_DECLS;
     }
 
-    bool foundExternBlock = false; // ignored
-    auto helper = LookupHelper(context, resolving, checkedScopes, result,
-                               foundExternBlock,
-                               /* traceCurPath */ nullptr,
-                               /* traceResult */ nullptr);
-    got = helper.doLookupInScope(scope, {}, name, config);
+    got = helpLookupInScopeWithShadowingWarning(
+                            context, scope, /* receiverScopes */ {},
+                            resolving, name, config,
+                            checkedScopes, result,
+                            idForWarnings);
```

Comment:
Am I correct to assume that this is safe because `lookupInScopeViz` is specifically used for resolving visibility statements, which means that the warning won't be spuriously issued by other thing performing scope lookup? I'm assuming we don't want the warning always since many things in the language internally perform searches.

Reply:
Right, that's the idea. We want to focus the warning on some particular cases even though other parts of the compiler implementation might occasionally do things like ask "Is Symbol XYZ available here?". As far as I know, the `lookupInScopeViz` case is only ever used for visibility statements. We could rename it something like `lookupInScopeVizWithWarning` if we wanted.